plugin/secrets_service_account_key.go

```diff
@@ -201,6 +281,10 @@ func (b *backend) getSecretKey(ctx context.Context, s logical.Storage, rs *RoleS
 		return logical.ErrorResponse(err.Error()), nil
 	}
 
+	if err := createCacheCollection(ctx, s, rs, key, ttlToUse); err != nil {
```

Comment:
To minimize the chance of creating orphan SAKs, I think we should call `createCacheCollection()` first, create the SAK, and then persist it to the `cacheCollection`.

Also, the SAK should be revoked when `cacheCollection.putToStorage()` returns an error. Since `secretKeyRevoke()` will throw an error if the `cacheCollection` is not available, we should not continue with the key creation flow if we can't put the SAK on the cache.

Reply:
> To minimize the chance of creating orphan SAKs, I think we should call createCacheCollection() first, create the SAK, and then persist it to the cacheCollection.
I think we can't `createCacheCollection()` before creating the SAK as the createCacheCollection requires the generated SAK 

> Also, the SAK should be revoked when cacheCollection.putToStorage() returns an error. Since secretKeyRevoke() will throw an error if the cacheCollection is not available, we should not continue with the key creation flow if we can't put the SAK on the cache.
Noted, I'll update it in the next commit.